pkg/providers/imagefamily/bootstrap/aksbootstrap.go

```diff
@@ -453,7 +455,10 @@ func (a AKS) applyOptions(nbv *NodeBootstrapVariables) {
 	nbv.ResourceGroup = a.ResourceGroup
 	nbv.UserAssignedIdentityID = a.KubeletIdentityClientID
 
-	nbv.NetworkPlugin = a.NetworkPlugin
+	// if we are using overlay, we delegate the installation of the cni to the cns daemonset and will put none as to not drop the
+	// 10-azure.conflist that will use nodesubnet
+	nbv.NetworkPlugin = lo.Ternary(a.NetworkPluginMode == consts.NetworkPluginModeOverlay, consts.NetworkPluginNone, a.NetworkPlugin)
```

Comment:
Please explicitely test for (Plugin == Azure && PluginMode = Overlay)
And let's do the same for Dataplane == Cilium

Reply:
Added tests

Reply:
Sorry, I did not make myself clear. I think
* _the check here_ needs to include `a.NetworkPlugin == const.NetworkPluginAzure` condition
* there should be another separate check for Dataplane == Cilium - which should also result in NetworkPlugin none

Reply:
yes to what Alex said.

The plan is that by k8s 1.32+ the logic will simplify to: if Azure CNI -> CNS configures the CNI (= the node has networkPlugin=none), for all variations of Azure CNI. We're not quite there yet though.